cli/pcluster/config/config_patch.py

```diff
@@ -0,0 +1,296 @@
+          same number of sections and the same sections labels
+        - the comparation step, when all sections of the two configurations are compared one by one and a report with
+          all found changes is generated
+
+    Once created, a ConfigPatch can be checked against an existing CloudFormation stack using the check() method. The
+    output of a check will contain the following information:
+        - A boolean telling if the new configuration can be safely applied
+        - The list of detected parameter changes
+        - The list of actions needed (if any) to unlock the update
+    """
+
+    IGNORED_SECTIONS = ["global", "aliases"]  # Sections ignored for patch creation
+
+    def __init__(self, base_config, target_config):
+        """
+        Create a ConfigPatch.
+
+        :param base_config: The base configuration, f.i. as reconstructed from CloudFormation
+        :param target_config: The target configuration, f.i. as loaded from configuration file
+        """
+        # Cached condition results
+        self.condition_results = {}
+
+        self.base_config = copy.deepcopy(base_config)
+        self.target_config = copy.deepcopy(target_config)
+        self.changes = []
+        self._adapt()
+        self._compare()
+
+    def _adapt(self):
+        """
+        Adapt the base config to the target one.
+
+        The adaptation phase involves restoring sections labels in the base config to make them match the ones in the
+        target config, plus adding missing sections into one or the other configuration to make comparison possible.
+        At the end of this process, both configurations will have the same number of sections, as well as the same
+        section labels.
+        """
+        # Remove ignored sections
+        self._remove_ignored_sections(self.base_config)
+        self._remove_ignored_sections(self.target_config)
+
+        #  Rename base section labels to prevent overlapping with target labels.
+        self._rename_base_sections()
+
+        # Adaptation: base -> target
+        self._adapt_from_base()
+        # Adaption: target -> base
+        self._adapt_from_target()
+
+    def _adapt_from_base(self):
+        """
+        For each base section, find the matching one in target configuration (or create it if missing).
+
+        To mark correspondence, assign the same section to the base section as the target section
+        """
+        for section_key in self.base_config.get_section_keys():
+            for _, base_section in self.base_config.get_sections(section_key).items():
+                target_section = self._get_target_section(base_section)
+                # Assign to base config's section the same label as the target section
+                base_section.label = target_section.label
+                LOGGER.debug(
+                    "Matched base section {0} {1} with target section {0} {2}".format(
```

Comment:
Instead of using {0} in this case we can assign a name to the variable:
```python
"Matched base section {key} {base_label} with target section {key} {target_label}".format(
    key=section_key, 
    base_label=base_section.label, 
    target_label=target_section.label
)
```

Reply:
Then I'd put the section name into square brackets, to simplify readability of the debug message:
`Matched base section [{key} {base_label}] with target section [{key} {target_label}]`